marklogic-data-hub/src/main/java/com/marklogic/hub/deploy/commands/UpdateDhsModulesPermissionsCommand.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2012-2019 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.marklogic.hub.deploy.commands;
+
+import com.marklogic.appdeployer.command.AbstractCommand;
+import com.marklogic.appdeployer.command.CommandContext;
+import com.marklogic.appdeployer.command.SortOrderConstants;
+import com.marklogic.client.ext.modulesloader.Modules;
+import com.marklogic.client.ext.modulesloader.impl.DefaultModulesFinder;
+import com.marklogic.hub.HubConfig;
+import com.marklogic.hub.dataservices.UpdateDhsResourcePermissions;
+import org.apache.commons.io.FilenameUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+@Component
+public class UpdateDhsModulesPermissionsCommand extends AbstractCommand {
```

Comment:
This command is specific to installing DHF into DHS, so a few notes on it:

- It should be in the com.marklogic.hub.cli.deploy package along with the other 2 commands specific to installing DHF into DHS
- It shouldn't be a Spring component - InstallIntoDhsCommand should just pass a HubConfig into its constructor (similar to CopyQueryOptionsCommand)

And then regarding the use of data services - I don't think we should do that for a deployment concern. CopyQueryOptionsCommand is using the /v1/eval endpoint to mess around with query options as part of installing DHF into DHS. For consistency and simplicity, I think this command should follow the same approach - i.e. build a script and send it to /v1/eval. Building a script inside of Java code isn't the cleanest thing ever, but I think it's suitable for a deployment concern - whereas Data Services is absolutely the best approach for any interface that we want a customer to be able to invoke (but we definitely don't want a customer invoking this endpoint).